samples/tutorials/DSL - Control structures/DSL - Control structures.py

```diff
@@ -49,75 +41,63 @@ def flip_coin_op() -> str:
 
+@dsl.component
+def fail_op(message: str):
+    """Fails."""
+    import sys
+    print(message)
+    sys.exit(1)
 
-# Submit the pipeline for execution:
-#kfp.Client(host=kfp_endpoint).create_run_from_pipeline_func(flipcoin_pipeline, arguments={})
 
 # %% [markdown]
+# ## Parallel execution
+# You can use the `with dsl.ParallelFor(task1.outputs) as items:` context to execute tasks in parallel
+
+# ## Conditional execution
+# You can use the `with dsl.Condition(task1.outputs["output_name"] = "value"):` context to execute parts of the pipeline conditionally
+
 # ## Exit handlers
 # You can use `with dsl.ExitHandler(exit_task):` context to execute a task when the rest of the pipeline finishes (succeeds or fails)
 
 # %%
-@func_to_container_op
-def fail_op(message):
-    """Fails."""
-    import sys
-    print(message)    
-    sys.exit(1)
 
 
 @dsl.pipeline(
-    name='Conditional execution pipeline with exit handler',
-    description='Shows how to use dsl.Condition() and dsl.ExitHandler().'
+    name='tutorial-control-flows',
+    description='Shows how to use dsl.Condition(), dsl.ParallelFor, and dsl.ExitHandler().'
 )
-def flipcoin_exit_pipeline():
-    exit_task = print_op('Exit handler has worked!')
+def control_flows_pipeline():
+    exit_task = print_op(message='Exit handler has worked!')
     with dsl.ExitHandler(exit_task):
-        flip = flip_coin_op()
-        with dsl.Condition(flip.output == 'heads'):
-            random_num_head = get_random_int_op(0, 9)
+        fail_op(
+            message="Failing the run to demonstrate that exit handler still gets executed."
+        )
+
+    flip = flip_coin_op()
+
+    with dsl.ParallelFor(['heads', 'tails']) as expected_result:
+
+        with dsl.Condition(flip.output == expected_result):
+            random_num_head = get_random_int_op(minimum=0, maximum=9)
             with dsl.Condition(random_num_head.output > 5):
-                print_op('heads and %s > 5!' % random_num_head.output)
+                print_op(
+                    message=f'{expected_result} and {random_num_head.output} > 5!'
+                )
             with dsl.Condition(random_num_head.output <= 5):
-                print_op('heads and %s <= 5!' % random_num_head.output)
-
-        with dsl.Condition(flip.output == 'tails'):
-            random_num_tail = get_random_int_op(10, 19)
-            with dsl.Condition(random_num_tail.output > 15):
-                print_op('tails and %s > 15!' % random_num_tail.output)
-            with dsl.Condition(random_num_tail.output <= 15):
-                print_op('tails and %s <= 15!' % random_num_tail.output)
+                print_op(
+                    message=f'{expected_result} and {random_num_head.output} <= 5!'
+                )
 
-        with dsl.Condition(flip.output == 'tails'):
-            fail_op(message="Failing the run to demonstrate that exit handler still gets executed.")
+        # with dsl.Condition(expected_result == 'tails'):
```

Comment:
It looks like we need to add a situation where the task fails so we can run the exit handler.